components/navigation/DesktopNav.tsx

```diff
@@ -9,8 +11,91 @@ import SearchRouteIcon from '../icons/navigation/SearchRouteIcon';
 import ProfileRouteIcon from '../icons/navigation/ProfileRouteIcon';
 import MessageRouteIcon from '../icons/navigation/MessageRouteIcon';
 
+import { useEffect, useState } from 'react';
+import selectUserUnreadConversations from '@/supabase/models/messaging/selectUserUnreadMessages';
+import { createClientComponentClient } from '@supabase/auth-helpers-nextjs';
+
 const DesktopNav = () => {
   const pathname = usePathname();
+  const supabase = createClientComponentClient();
+  const [userId, setUserId] = useState<string>('');
+  const [notification, setNotification] = useState<boolean>(false);
+  const [trigger, setTrigger] = useState<number>(0);
+
+  useEffect(() => {
+    const getUserSession = async () => {
+      try {
+        const { data: userData } = await supabase.auth.getSession();
+        if (userData.session?.user.id) {
+          setUserId(userData.session?.user.id);
+        }
+      } catch (error) {
+        console.error('Error getting user session: ', error);
+      }
+    };
+    getUserSession();
```

Comment:
So the code for this `useEffect` looks like this: 
```
  useEffect(() => {
    const getUserId = async () => {
      const data = await selectLoggedUserId();
      if (data) {
        setUserId(data);
      } else {
        console.error('Failed to fetch user Id');
      }
    };
    getUserId();
  }, []);
```
I removed curly brackets on `data` variable because I assume we only want to receive `userId` and nothing else from `selectLoggedUserId` function and I still want to log some kind of `error` if their is no data fetched in `selectLoggedUserId` function there is error handling too.

```
import { createClientComponentClient } from '@supabase/auth-helpers-nextjs';

export default async function selectLoggedUserId() {
  const supabase = createClientComponentClient();
  try {
    const { data, error } = await supabase.auth.getSession();

    if (error) throw error;

    return data.session?.user.id;
  } catch (error) {
    console.error('Error getting user session: ', error);
  }
}
```